app/WebParsing/ReqParser.hs

```diff
@@ -135,14 +135,47 @@ singleParser = do
     num <- Parsec.count 3 Parsec.digit
     -- TODO: Make the last two letters more restricted.
     sess <- Parsec.count 2 Parsec.alphaNum
-    return $ J (code ++ num ++ sess)
+    return $ J (code ++ num ++ sess) ""
+
+-- | Parser for single courses or "atomic" Reqs represented by a J.
+justParser :: Parser Req
+justParser = do
+    code <- Parsec.count 3 Parsec.letter
+    num <- Parsec.count 3 Parsec.digit
+    sess <- Parsec.count 2 Parsec.alphaNum
+    Parsec.spaces
+    lpar <- Parsec.option "" $ Parsec.string "("
```

Comment:
This can be improved if you use the `between` combinator. Something like

`meta <- Parsec.option "" $ Parsec.between lParen rParen markInfoParser`